### ISSUE TITLE
HF 15 Sim doesn't work

### ISSUE THREAD

**Issue by mubix** [Proxmark/proxmark3#20]
```
{"sim", CmdHF15Sim, 0, "Fake an ISO15693 tag"},
```

But 

```
// Simulation is still not working very good
int CmdHF15Sim(const char *Cmd)
{
        UsbCommand c = {CMD_SIMTAG_ISO_15693, {strtol(Cmd, NULL, 0), 0, 0}};
        SendCommand(&c);
        return 0;
}
```

I get weird results, like it'll out put something like the following (3 different runs):

```
1K08080000*
4K04000400*
4K00100010*
```

**Comment by iceman1001** [comment]
Looking at the iso15693.c code, I realise that it needs to be fix to handle @pwpiwi 's new BigBuff management.

**Comment by douniwan5788** [comment]
Just in case, I hope this helps .
I have successfully simulated a 15693tag identical with NXP ICODE2 , but messed up all the other functions.
https://github.com/douniwan5788/proxmark3/commits/newWorking15693sim

**Comment by sahilsaini65** [comment]
Hi @douniwan5788 . I tried your code from https://github.com/douniwan5788/proxmark3/commits/newWorking15693sim, however i am still not able to simulate an ISO15693 tag. I am trying to simulate an SL2S5402 tag. Any help?

**Comment by jbarg** [comment]
are there any news of a working ISO15693 simulation? i can't get it to work :(

**Comment by iceman1001** [comment]
Haven't heard of someone working on it.  Have you tried @douniwan5788 's fork?

**Comment by jbarg** [comment]
Yes, i have a bootloop with his image though :/ used an old 4.8 gcc to compile it. code breaks with newer compilers

**Comment by iceman1001** [comment]
if you compare the crapto code that breaks with the new code, you can see what you'll need to change to make it compile.

**Comment by douniwan5788** [comment]
Please use this commit
https://github.com/douniwan5788/proxmark3/commit/01c9af3f7420593c7a45affb6e3ee143213e3fcd

Sorry for mentioned that two subcarrier is not working at all.

**Comment by iceman1001** [comment]
@douniwan5788  Which xilinix version do you use to compile the FPGA code?

**Comment by jbarg** [comment]
@douniwan5788  @iceman1001  Thanks guys. works now.

1. used an old 4.8 gcc to compile
2. flashed bootloader and image
3. works (simulating doesnt work, i'm currently writing my block modifications on the card with hf 15 cmd write)

Thanks for the help!

**Comment by m2otech** [comment]
@douniwan5788 

I used your commit linked above to try to emulate 15693 tags. It compiled fine but when running:

hf 15 sim E0xxxxxx (all 16 bytes)

I get the error "vicc_memory_map.bin"

Do I need to read a tag first and save it in this file?

Any other hint where I could look to fix this? Has anyone been able to emulate 15693?

**Comment by iceman1001** [comment]
Does  **hf1 15 sim** work now with @pwpiwi  fix to iso15693 commands?

**Comment by pwpiwi** [comment]
Should be unrelated. My fix was for the FPGA code used by the PM when acting AS reader.

**Comment by iceman1001** [comment]
then we keep this issue open.

**Comment by hduarte** [comment]
I've tried it with iceman version:
 [ ARM ]
 bootrom: iceman/master/ice_v3.1.0-949-gdb7cb418 2018-07-17 19:57:26
      os: iceman/master/ice_v3.1.0-949-gdb7cb418 2018-07-17 19:57:28
 [ FPGA ]
 LF image built for 2s30vq100 on 2017/10/25 at 19:50:50
 HF image built for 2s30vq100 on 2017/11/10 at 19:24:16

and after running:

pm3 --> hf 15 sim E016240000000000
Starting simulating UID E0 16 24 00 00 00 00 00           
#db# ISO-15963 Simulating uid: E016240000000000

nothing is picked up but the reader

**Comment by pwpiwi** [comment]
Fixed with PR #696. Please note that hf 15 sim functionality is still limited to the inventory command response yet.
